src/target/riscv/riscv-013.c

```diff
@@ -691,16 +694,19 @@ static riscv_reg_t read_abstract_arg(struct target *target, unsigned index,
 		unsigned size_bits)
 {
 	riscv_reg_t value = 0;
+	uint32_t v;
 	unsigned offset = index * size_bits / 32;
 	switch (size_bits) {
 		default:
 			LOG_ERROR("Unsupported size: %d", size_bits);
 			return ~0;
 		case 64:
-			value |= ((uint64_t) dmi_read(target, DMI_DATA0 + offset + 1)) << 32;
+			dmi_read(target, &v, DMI_DATA0 + offset + 1);
```

Comment:
your PR notes that you propogate the error from `dmi_read` everywhere, but here you don't check it?

Reply:
This function does not return an error code.